doc/tutorials/using_ubuntu_as_sos.rst

```diff
@@ -109,42 +109,57 @@ the source code, build it, and install it on your device.
 
          sudo efibootmgr -c -l "\EFI\acrn\acrn.efi" -d /dev/sda -p 1 \
                 -L "ACRN Hypervisor" -u "bootloader=\EFI\ubuntu\grubx64.efi"
-         # Verify that the "ACRN Hypervisor" will be booted first
-         sudo efibootmgr -v
+         
+   # Verify that the "ACRN Hypervisor" and make sure it will be booted first
+   .. code-block:: none
+      
+      sudo efibootmgr -v
+
+   # You can change the boot order at any time using
+   .. code-block:: none
+   
+      efibootmgr -o XXX,XXX,XXX
 
-      You can change the boot order at any time using ``efibootmgr
-      -o XXX,XXX,XXX``
+    .. note::
+       By default, the “ACRN Hypervisor” you have just added should be
+       the first one to boot. Verify this by using ``efibootmgr -v`` or
+       by entering the EFI firmware at boot (using :kbd:`F10`)
 
-      .. note::
-         By default, the “ACRN Hypervisor” you have just added should be
-         the first one to boot. Verify this by using ``efibootmgr -v`` or
-         by entering the EFI firmware at boot (using :kbd:`F10`)
 
 Install the Service OS kernel
 *****************************
 
-You can download latest Service OS kernel from
+You can download Service OS kernel from clear Linux project: https://download.clearlinux.org/releases
+
+1, Download latest Service OS kernel from latest Clear Linux releasee is here:
 `<https://download.clearlinux.org/releases/current/clear/x86_64/os/Packages/>`_
-(**We need to update the URL to one that is known to work, and matches the
-tag/release we use above!!!**). We will extract it and install it in the next
-few steps.
+linux-pk414-sos-<``kernel-version``>-<``build-version``>.x86_64.rpm
 
-a. Download and extract the Service OS kernel
+2, Download Specific Service OS kenrel from specific Clear Linux release:
+https://download.clearlinx.org/release/<Clear-linux-release>/clear/x86_64/os/Packages
+for example:
+Clear-linux-release = 24040
+https://download.clearlinux.org/releases/24040/clear/x86_64/os/linux-pk414-sos-4.14.57-69.x86_64.rpm
+
+.. note:: 
+       in this guide, we use and recommend people downloading latest release from "current" path
+
+a. Download and extract the latest Service OS kernel(for example: current=24040)
 
    .. code-block:: none
 
       mkdir ~/kernel-build
       cd ~/kernel-build
-      wget  https://download.clearlinux.org/releases/current/clear/x86_64/os/Packages/linux-pk414-sos-4.14.41-39.x86_64.rpm
+      wget  https://download.clearlinux.org/releases/24040/clear/x86_64/os/Packages/linux-pk414-sos-4.14.57-69.x86_64.rpm
```

Comment:
I'm confused.  You just recommended they download from the ``current`` folder, but here you're using the ``24040`` folder?